src/optimagic/deprecations.py

```diff
@@ -95,3 +96,33 @@ def replace_and_warn_about_deprecated_algo_options(algo_options):
         out[replacements[k]] = algo_options[k]
 
     return out
+
+
+def replace_and_warn_about_deprecated_bounds(
+    lower_bounds,
+    upper_bounds,
+    bounds,
+    soft_lower_bounds=None,
+    soft_upper_bounds=None,
+):
+    old_bounds = {
+        "lower": lower_bounds,
+        "upper": upper_bounds,
+        "soft_lower": soft_lower_bounds,
+        "soft_upper": soft_upper_bounds,
+    }
+
+    old_present = [k for k, v in old_bounds.items() if v is not None]
+
+    if old_present:
+        msg = (
+            f"Specifying bounds via the arguments {', '.join(old_present)} is "
+            "deprecated and will be removed in optimagic version 0.6.0 and later. "
+            "Please use the `bounds` argument instead."
+        )
+        warnings.warn(msg, FutureWarning)
+
+    if bounds is None and old_present:
+        bounds = Bounds(**old_bounds)
```

Comment:
What about the case where both the new and old bounds are set. In the current implementation we use the new bounds, but do not throw a warning other than the FutureWarning above.

Reply:
As discussed, this behavior is safe enough.

Reply:
Same.